packages/plugin-awesome/test/fixtures.ts

```diff
@@ -0,0 +1,199 @@
+    hidden: false,
+    labels: [],
+    parameters: [],
+    links: [],
+    attachments: [],
+    steps: [],
+    retries: [],
+    history: [],
+    retry: false,
+    flaky: false,
+    fullName: "Failed test",
+    description: "",
+    descriptionHtml: "",
+    statusDetails: {},
+    testCaseId: "failed-test",
+    historyId: "failed-test",
+  },
+  {
+    id: "broken-test",
+    name: "Broken test",
+    status: "broken",
+    start: 0,
+    stop: 0,
+    hidden: false,
+    labels: [],
+    parameters: [],
+    links: [],
+    attachments: [],
+    steps: [],
+    retries: [],
+    history: [],
+    retry: false,
+    flaky: false,
+    fullName: "Broken test",
+    description: "",
+    descriptionHtml: "",
+    statusDetails: {},
+    testCaseId: "broken-test",
+    historyId: "broken-test",
+  },
+  {
+    id: "skipped-test",
+    name: "Skipped test",
+    status: "skipped",
+    start: 0,
+    stop: 0,
+    hidden: false,
+    labels: [],
+    parameters: [],
+    links: [],
+    attachments: [],
+    steps: [],
+    retries: [],
+    history: [],
+    retry: false,
+    flaky: false,
+    fullName: "Skipped test",
+    description: "",
+    descriptionHtml: "",
+    statusDetails: {},
+    testCaseId: "skipped-test",
+    historyId: "skipped-test",
+  },
+  {
+    id: "unknown-test",
+    name: "Unknown test",
+    status: "unknown",
+    start: 0,
+    stop: 0,
+    hidden: false,
+    labels: [],
+    parameters: [],
+    links: [],
+    attachments: [],
+    steps: [],
+    retries: [],
+    history: [],
+    retry: false,
+    flaky: false,
+    fullName: "Unknown test",
+    description: "",
+    descriptionHtml: "",
+    statusDetails: {},
+    testCaseId: "unknown-test",
+    historyId: "unknown-test",
+  },
+];
+
+const store: AllureStore = {
+  allTestResults: async () => testResults,
+  fixturesByTrId: async () => [],
+  historyByTrId: async () => [],
+  retriesByTrId: async () => [],
+  attachmentsByTrId: async () => [],
+  allTestEnvGroups: async () => [],
+  allEnvironments: async () => [],
+  allVariables: async () => [],
+  envVariables: async () => [],
+  metadataByKey: async () => [],
+  allAttachments: async () => [mockAttachment],
+  attachmentContentById: async () => mockAttachment,
+  testsStatistic: async () => ({
+    total: testResults.length,
+    passed: testResults.filter((tr) => tr.status === "passed").length,
+    failed: testResults.filter((tr) => tr.status === "failed").length,
+    broken: testResults.filter((tr) => tr.status === "broken").length,
+    skipped: testResults.filter((tr) => tr.status === "skipped").length,
+    unknown: testResults.filter((tr) => tr.status === "unknown").length,
+  }),
+  allHistoryDataPoints: async () => [],
+};
+
+const context: PluginContext = {
+  reportFiles: {
+    addFile: async () => {},
+  },
+};
+
+export const fixtures = {
+  store,
+  context,
+};
+
+export function createFilesCollectorWithCheck(check: (path: string, content?: Buffer) => void) {
```

Comment:
It's not a `FilesCollector`, it's a `ReportFiles` abstraction we have defined an interface for.

I propose not to create additional abstraction but use a mock instead:

```ts
import { vi, type MockedFunction } from "vitest";
import type { ReportFiles } from "@allurereport/plugin-api";

const mockReportFiles = {
  addFile: vi.fn(),
} as ReportFiles
const reportDataFileWriter = new ReportFileDataWriter(mockReportFiles)

reportDataFileWriter.writeData("path/to/my/data/file.json", "data")

// then check, that mocked method has been called with the desired parameters
expect(mockReportFiles.addFile as MockedFunction<any>).toHaveBeenCalledTimes(1);
expect(mockReportFiles.addFile as MockedFunction<any>).toHaveBeenCalledNthWith(["path/to/my/data/file.json", "data"]);
```